### PR TITLE
Add new GCC built-in definitions

### PR DESCRIPTION
**Pull request by StefanosChaliasos**
Make `_Static_assert` immutable, and add definitions for
the following macros.

```
__builtin_add_overflow_p
__builtin_mul_overflow_p
__builtin_sub_overflow
__builtin_sub_overflow_p
```

<https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html>

### CODE REVIEW

File: include/template/gcc-defs.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #define _Alignof(x) (sizeof(x) & 0xf)
 #define __attribute__(x)
 #define __builtin_add_overflow(x,y,z) ((x), (y), (z), 1)
+#define __builtin_add_overflow_p(x,y,z) ((x), (y), (z))
```
Comment (dspinellis) [inline include/template/gcc-defs.h:34]: It's better to use 1 as the result. The value of z isn't used for any other reason than check its type.

File: include/template/gcc-defs.h [hunk 2]
```diff
@@ -87,6 +88,7 @@
 #define __builtin_memset(x,y,z) ((x), (y), (z), 1)
 #define __builtin___memset_chk(x,y,z,w) ((x), (y), (z), (w), 0)
 #define __builtin_mul_overflow(x,y,z) ((x), (y), (z), 1)
+#define __builtin_mul_overflow_p(x,y,z) ((x), (y), (z))
```
Comment (dspinellis) [inline include/template/gcc-defs.h:91]: Ditto.

File: include/template/gcc-defs.h [hunk 3]
```diff
@@ -126,6 +128,8 @@
 #define __builtin_strlen(x) ((x), 1)
 #define __builtin___strncat_chk(x,y,z,w) ((x), (y), (z), (w), 1)
 #define __builtin___strncpy_chk(x,y,z,w) ((x), (y), (z), (w), 1)
+#define __builtin_sub_overflow(x,y,z) ((x), (y), (z), 1)
+#define __builtin_sub_overflow_p(x,y,z) ((x), (y), (z))
```
Comment (dspinellis) [inline include/template/gcc-defs.h:132]: Ditto.

File: include/template/gcc-defs.h (hunk 1 above)
Comment (StefanosChaliasos) [reply include/template/gcc-defs.h]: Done

**Pull request merged**